Jint 3.x can not work in unity

there is an error when import the jint@latest in nuget.

Assets/NewBehaviourScript.cs(23,9): error CS0012: The type 'Script' is defined in an assembly that is not referenced. You must add a reference to assembly 'Esprima, Version=2.1.2.0, Culture=neutral, PublicKeyToken=2e92ba9c8d81157f'.

Jint Version: jint.3.0.0-beta-2038
Unity Version: 2021.3.2f1

Hello,
I'm also using Jint in a Unity project and updated Jint today to the newest version. Now I get a similiar message:

1. I'm adding [Jint 3.0.0-beta-2043](https://www.nuget.org/packages/Jint/3.0.0-beta-2043) to my projekt. After importing the new DLL I get for every jint-line in my project this message
`error CS0012: The type 'Script' is defined in an assembly that is not referenced. You must add a reference to assembly 'Esprima, Version=3.0.0.0, Culture=neutral, PublicKeyToken=2e92ba9c8d81157f'.`

2. I 'm adding [Esprima 3.0.0-beta-9](https://www.nuget.org/packages/Esprima/3.0.0-beta-9#versions-body-tab). After importing that I get the next message
`Error: Could not load signature of HexConverter:get_CharToHexLookup due to: Could not load file or assembly 'System.Memory, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. assembly:System.Memory, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 type:<unknown type> member:(null) signature:<none>`

@GDdark How did you fix your problem? Did it run after importing Esprima?
@lahma Have you already faced this problem?

--- a/Jint.Tests/Runtime/AsyncTests.cs
+++ b/Jint.Tests/Runtime/AsyncTests.cs
@@ -2108,6 +2108,26 @@ public class AsyncTests
         Assert.Equal("10,20,30", result.AsString());
     }
 
+    [Fact]
+    public void AwaitInsideForOfLoopShouldWorkWithHeadDestructuring()
+    {
+        var engine = new Engine();
+        var result = engine.Evaluate("""
+            (async function() {
+                var results = [];
+                var items = { a: 1, b: 2 };
+                for (const [key, value] of Object.entries(items)) {
+                    var r = await Promise.resolve(value * 10);
+                    results.push(key + ":" + r);
+                }
+                return results.join(",");
+            })()
+            """);
+
+        result = result.UnwrapIfPromise();
+        Assert.Equal("a:10,b:20", result.AsString());
+    }
+
     class TestAsyncClass
     {
         private readonly ConcurrentBag<string> _values = new();

--- a/Jint/Runtime/Interpreter/Statements/JintForInForOfStatement.cs
+++ b/Jint/Runtime/Interpreter/Statements/JintForInForOfStatement.cs
@@ -455,6 +455,7 @@ internal sealed class JintForInForOfStatement : JintStatement<Statement>
                     context.Engine.Debugger.OnStep(_leftNode);
                 }
 
+                var valueForResume = nextValue;
                 var status = CompletionType.Normal;
                 if (!destructuring)
                 {
@@ -482,13 +483,10 @@ internal sealed class JintForInForOfStatement : JintStatement<Statement>
                 }
                 else
                 {
-                    // Save the original iterator value before destructuring (for potential resume)
-                    var iteratorValue = nextValue;
-
                     nextValue = DestructuringPatternAssignmentExpression.ProcessPatterns(
                         context,
                         _assignmentPattern!,
-                        iteratorValue,
+                        valueForResume,
                         iterationEnv,
                         checkPatternPropertyReference: _lhsKind != LhsKind.VarBinding);
 
@@ -500,7 +498,7 @@ internal sealed class JintForInForOfStatement : JintStatement<Statement>
                         if (_iterationKind == IterationKind.AsyncIterate && suspendable is not null)
                         {
                             var asyncSD = suspendable.Data.GetOrCreate<ForAwaitSuspendData>(this);
-                            asyncSD.CurrentValue = iteratorValue;
+                            asyncSD.CurrentValue = valueForResume;
                             asyncSD.AccumulatedValue = v;
                         }
                         completionType = CompletionType.Return;
@@ -561,7 +559,7 @@ internal sealed class JintForInForOfStatement : JintStatement<Statement>
                 {
                     var data = generator.Data.GetOrCreate<ForOfSuspendData>(this, iteratorRecord);
                     data.AccumulatedValue = v;
-                    data.CurrentValue = nextValue;
+                    data.CurrentValue = valueForResume;
                     data.IterationEnv = iterationEnv;
                     data.OuterEnv = oldEnv;
                 }
@@ -574,7 +572,7 @@ internal sealed class JintForInForOfStatement : JintStatement<Statement>
                 {
                     var asyncData = asyncFnBody.Data.GetOrCreate<ForOfSuspendData>(this, iteratorRecord);
                     asyncData.AccumulatedValue = v;
-                    asyncData.CurrentValue = nextValue;
+                    asyncData.CurrentValue = valueForResume;
                     asyncData.IterationEnv = iterationEnv;
                     asyncData.OuterEnv = oldEnv;
                 }

